UofT-DSI | deep_learning - Assignment 2

Zenduxi opened the pull request:
Changes to be committed:
	modified:   01_materials/labs/lab_3.ipynb
	modified:   01_materials/labs/lab_4.ipynb
	modified:   01_materials/labs/lab_5.ipynb
	modified:   01_materials/labs/lab_6.ipynb
	modified:   02_activities/assignments/assignment_2.ipynb

## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Add code to train and evaluate different models like CNN and zero-shot tranformers.

## What did you learn from the changes you have made?
I learnt building a recommender system, processing images with CNN models, object classification and localization with ResNet, text classification with Neural Network models in tensorflow.keras, and Zero-Shot Image Classification with Transformers.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I'd like to try different models or different parameters in the same model. More exploration will be done in the near future.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I need to spend much time after classes learning the models and learn the code to implement them.

## How were these changes tested?
Have to work harder :-) Also review the recordings.

## A reference to a related issue in your repository (if applicable)

## Checklist
- [ X] I can confirm that my changes are working as intended

Zenduxi committed:
On branch assignment-2
Changes to be committed:
	modified:   01_materials/labs/lab_3.ipynb
	modified:   01_materials/labs/lab_4.ipynb
	modified:   01_materials/labs/lab_5.ipynb
	modified:   01_materials/labs/lab_6.ipynb
	modified:   02_activities/assignments/assignment_2.ipynb
Add code to train and evaluate different models like CNN and zero-shot tranformers.

dwz92 reviewed (changes requested):
I see that your labs are complete, but I am unable to load your a2. Please ensure the file is not corrupted and resubmit

dwz92 reviewed (approved):
Reflcetions, working solutions, and submitted labs

Zenduxi committed:
Changes to be committed:
	modified:   02_activities/assignments/assignment_2.ipynb
Coding and writing reflections & the short report following the requirements.

Zenduxi commented:
@dwz92 
Hi Emma, sorry for reading your message later. 
Apologize that I didn't verify the loading of the notebook file on Github. Were you able to view it with some workaround? 
I just tried https://nbviewer.org and afer providing the following url of my assignment_2.ipynb file, I can view it on the website. I tried resubmitting and that didn't help. Probably I need to clear some cell outputs.
https://github.com/Zenduxi/deep_learning/blob/783eef61f554d38f6636dd5eb01f9a1854c02487/02_activities/assignments/assignment_2.ipynb
